### PR TITLE
OracleUI

### PR DESCRIPTION
**Pull request by ike709**
## About The Pull Request
Ports OracleUI and associated changes from my Interstation13 port. I haven't had time to test this code.

## Why It's Good For The Game
TGUI is slow and it's a real pain to work with. OracleUI isn't.

## Changelog
:cl: Oracle, ported by ike709
add: OracleUI
tweak: Paper, books, and disposal units have been converted to OracleUI.
add: Nanotrasen and Syndicate stamps.
/:cl:

### CODE REVIEW

File: code/modules/oracle_ui/themed.dm [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		return errormsg
+
+/datum/oracle_ui/themed/proc/get_content_file(filename)
+	return get_file("html/oracle_ui/content/[content_root]/[filename]")
+
+/datum/oracle_ui/themed/proc/get_themed_file(filename)
+	return get_file("html/oracle_ui/themes/[theme]/[filename]")
+
+/datum/oracle_ui/themed/proc/process_template(template, variables)
+	var/regex/pattern = regex("\\@\\{(\\w+)\\}","gi")
```
Comment (ike709) [inline code/modules/oracle_ui/themed.dm:40]: This can probably be static.

**Comment by ike709** [comment]
Seems to survive basic testing, but I'd still like to see it testmerged first.

**Comment by ike709** [comment]
Wikibooks runtime and I haven't figured out why yet.

**Comment by qwertyquerty** [comment]
I can testmerge this soon

**Comment by qwertyquerty** [comment]
This looks really good. I can't believe it's actually fully documented. It looks really simple to port machines over. I also love the themes stuff. Although..

Shouldn't all atoms have a ui var instead of defining it where it's used?

**Pull request merged**